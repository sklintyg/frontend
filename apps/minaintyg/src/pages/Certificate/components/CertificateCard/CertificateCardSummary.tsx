import { useFormat } from '../../../../hooks/useFormat'
import { CertificateSummary } from '../../../../schema/certificate.schema'

export function CertificateCardSummary({ summary, timestamp }: { summary: CertificateSummary; timestamp: string }) {
  const { date } = useFormat()

  return (
    <div className="mb-2.5 flex flex-col place-content-end gap-2.5 border-stone-line md:mb-5 md:flex-row md:border-b md:pb-5">
      {summary && (
        <div className="grow leading-5">
          <h4 className="ids-heading-4 mb-0 leading-5 md:inline-block md:after:content-[':']">{summary.label}</h4> {summary.value}
        </div>
      )}
      <div className="flex flex-col leading-5 md:flex-row md:gap-1.5 whitespace-nowrap">
        <h4 className="ids-heading-4 mb-0 leading-5 md:inline-block md:after:content-[':']">Intyg utf√§rdat</h4>
        <span>{date(timestamp)}</span>
      </div>
    </div>
  )
}
