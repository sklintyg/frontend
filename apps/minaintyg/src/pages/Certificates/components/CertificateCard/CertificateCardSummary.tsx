import { useFormat } from '../../../../hooks/useFormat'
import { CertificateListSummary } from '../../../../schema/certificateList.schema'

export function CertificateCardSummary({ summary, timestamp }: { summary: CertificateListSummary; timestamp: string }) {
  const { formatDate } = useFormat()

  return (
    <div className="mb-2.5 flex flex-col place-content-end gap-2.5 md:mb-5 md:flex-row md:border-b md:pb-5">
      {summary.length > 0 && (
        <p className="grow">
          <span className="font-bold">Avser </span>
          {summary.map(([description, value]) => (
            <>
              <span className="font-bold">{description}:</span> {value}{' '}
            </>
          ))}
        </p>
      )}
      <div className="flex flex-col md:flex-row md:gap-1.5">
        <span className="font-bold">Intyg utf√§rdat:</span>
        <span>{formatDate(timestamp)}</span>
      </div>
    </div>
  )
}
