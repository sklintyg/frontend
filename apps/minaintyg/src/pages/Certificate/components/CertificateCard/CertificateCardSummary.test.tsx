import { render } from '@testing-library/react'
import { expect, it } from 'vitest'
import { CertificateCardSummary } from './CertificateCardSummary'

it('Should render correctly with summary', () => {
  const { container } = render(
    <CertificateCardSummary
      summary={{ label: 'GÃ¤ller intygsperiod', value: '2023-05-09 - 2023-06-23' }}
      timestamp="2023-05-09T18:37:23.159Z"
    />
  )
  expect(container).toMatchSnapshot()
})

it('Should render correctly without summary', () => {
  const { container } = render(<CertificateCardSummary summary={undefined} timestamp="2023-05-09T18:37:23.159Z" />)
  expect(container).toMatchSnapshot()
})
