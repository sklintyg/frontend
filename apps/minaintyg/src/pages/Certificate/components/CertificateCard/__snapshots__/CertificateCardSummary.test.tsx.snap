// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Should render correctly with summary 1`] = `
<div>
  <div
    class="mb-2.5 flex flex-col place-content-end gap-2.5 md:mb-5 md:flex-row md:border-b md:pb-5"
  >
    <p
      class="grow"
    >
      <span
        class="font-bold"
      >
        Gäller intygsperiod
        :
      </span>
       
      2023-05-09 - 2023-06-23
    </p>
    <div
      class="flex flex-col md:flex-row md:gap-1.5"
    >
      <span
        class="font-bold"
      >
        Intyg utfärdat:
      </span>
      <span>
        2023-05-09
      </span>
    </div>
  </div>
</div>
`;

exports[`Should render correctly without summary 1`] = `
<div>
  <div
    class="mb-2.5 flex flex-col place-content-end gap-2.5 md:mb-5 md:flex-row md:border-b md:pb-5"
  >
    <div
      class="flex flex-col md:flex-row md:gap-1.5"
    >
      <span
        class="font-bold"
      >
        Intyg utfärdat:
      </span>
      <span>
        2023-05-09
      </span>
    </div>
  </div>
</div>
`;
